core/src/Kokkos_Core_fwd.hpp

```diff
@@ -123,19 +123,15 @@ class OpenMPTargetSpace;
 }  // namespace Experimental
 #endif
 
-#if defined(KOKKOS_ENABLE_CUDA)
```

Comment:
Did you mean to provide the forward declaration when CUDA is not enabled?

Reply:
The other option is to sprinkle `#ifdef KOKKOS_ENABLE_CUDA` everywhere around the code where there's Cuda-specific behavior (instead of just using `if(std::is_same<execution_space, Kokkos::Cuda>::value)`)

Reply:
This is still a blocker.